R/target_market_share.R

```diff
@@ -402,8 +402,8 @@ calculate_ald_benchmark <- function(ald, region_isos, by_company) {
       .data$sector, .data$year, .data$region, .data$source
     ) %>%
     mutate(
-      .x = .data$production,
-      technology_share = .data$.x / sum(.data$.x),
+      .x = sum(.data$production),
```

Comment:
Every time I see sum I think of `na.rm`. Maybe `NA`s are handled above but one way to remove the worry is to either add `na.rm = TRUE` explicitely (if it corresponds) or maybe implement `add()` as a wrapper of `sum()` that defaults to `na.rm = TRUE`.

Reply:
I am a little hesitant to pass `na.rm = TRUE` to calls to `sum()`, my reasoning is as follows: 
#306 was flagged because `target_market_share` passing an error, due to some `NA` that was added internally. The input data had no `NA`s, so it was something in the code that was introducing the `NA`s. 

These `NA`s were not expected, so this error forced us to recognize the problem and address it. 

If we had passed `na.rm = TRUE`, I imagine this bug would have persisted silently, and we likely would have taken a lot longer to learn about it.